src/panels/config/automation/action/types/ha-automation-action-choose.ts

```diff
@@ -108,67 +99,99 @@ export class HaChooseAction extends LitElement implements ActionElement {
     const action = this.action;
 
     return html`
-      ${(action.choose ? ensureArray(action.choose) : []).map(
-        (option, idx) =>
-          html`<ha-card>
-            <ha-expansion-panel
-              leftChevron
-              @expanded-changed=${this._expandedChanged}
-            >
-              <h3 slot="header">
-                ${this.hass.localize(
-                  "ui.panel.config.automation.editor.actions.type.choose.option",
-                  "number",
-                  idx + 1
-                )}:
-                ${this._getDescription(option, idx)}
-              </h3>
-
-              <ha-icon-button
-                slot="icons"
-                .idx=${idx}
-                .disabled=${this.disabled}
-                @click=${this._removeOption}
-                .label=${this.hass.localize(
-                  "ui.panel.config.automation.editor.actions.type.choose.remove_option"
-                )}
-                .path=${mdiDelete}
-              ></ha-icon-button>
-              <div class="card-content">
-                <h4>
-                  ${this.hass.localize(
-                    "ui.panel.config.automation.editor.actions.type.choose.conditions"
-                  )}:
-                </h4>
-                <ha-automation-condition
-                  nested
-                  .conditions=${ensureArray<string | Condition>(
-                    option.conditions
-                  )}
-                  .reOrderMode=${this.reOrderMode}
-                  .disabled=${this.disabled}
-                  .hass=${this.hass}
-                  .idx=${idx}
-                  @value-changed=${this._conditionChanged}
-                ></ha-automation-condition>
-                <h4>
+      <div class="options">
+        ${repeat(
+          action.choose ? ensureArray(action.choose) : [],
+          (option) => this._getKey(option),
```

Comment:
This would do the same I think?
```suggestion
          (option) => option,
```

Reply:
You use the option as a key, but use a wrapper with a random string, my proposal also uses the option as a key, just without the string.

If this is used in other places I think we can clean it up there too.

Reply:
It seems to work fine without the custom key function. removed.